tests/unit/test_client.py

```diff
@@ -436,6 +436,54 @@ def test_get_bucket_with_string_hit(self):
             timeout=self._get_default_timeout(),
         )
 
+    def test_get_bucket_with_metageneration_match(self):
+        from google.cloud.storage.bucket import Bucket
+
+        PROJECT = "PROJECT"
+        CREDENTIALS = _make_credentials()
+        METAGENERATION_NUMBER = 6
+        client = self._make_one(project=PROJECT, credentials=CREDENTIALS)
+
+        BUCKET_NAME = "bucket-name"
+        URI1 = "/".join(
+            [
+                client._connection.API_BASE_URL,
+                "storage",
+                client._connection.API_VERSION,
+                "b",
+                "%s?projection=noAcl&ifMetagenerationMatch=%s"
+                % (BUCKET_NAME, METAGENERATION_NUMBER),
+            ]
+        )
+        URI2 = "/".join(
+            [
+                client._connection.API_BASE_URL,
+                "storage",
+                client._connection.API_VERSION,
+                "b",
+                "%s?ifMetagenerationMatch=%s&projection=noAcl"
+                % (BUCKET_NAME, METAGENERATION_NUMBER),
+            ]
+        )
```

Comment:
URI params order is not guaranted, so checking both variant. The alternative here is to add sorting somewhere in `JSONConnection.build_api_url()`